homeassistant/components/airpatrol/climate.py

```diff
@@ -0,0 +1,321 @@
+
+PARALLEL_UPDATES = 0
+
+_LOGGER = logging.getLogger(__name__)
+
+# HVAC modes supported by AirPatrol
+HVAC_MODES = {
+    "heat": HVACMode.HEAT,
+    "cool": HVACMode.COOL,
+    "off": HVACMode.OFF,
+}
+
+# Fan speeds supported by AirPatrol
+FAN_MODES = {
+    "min": FAN_LOW,
+    "max": FAN_HIGH,
+    "auto": FAN_AUTO,
+}
+
+# Swing modes supported by AirPatrol
+SWING_MODES = {
+    "on": SWING_ON,
+    "off": SWING_OFF,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up AirPatrol climate entities."""
+    coordinator: AirPatrolDataUpdateCoordinator = config_entry.runtime_data
+    # Create climate entities for each unit
+    entities = []
+    client = await coordinator.async_get_api_client()
+    units = await client.get_data()
+    entities = [
+        AirPatrolClimate(coordinator, unit, unit["unit_id"])
+        for unit in units
+        if "climate" in unit
+    ]
+
+    async_add_entities(entities)
+
+
+class AirPatrolClimate(
+    CoordinatorEntity[AirPatrolDataUpdateCoordinator], ClimateEntity
+):
+    """AirPatrol climate entity."""
+
+    _attr_has_entity_name = True
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.FAN_MODE
+        | ClimateEntityFeature.SWING_MODE
+        | ClimateEntityFeature.TURN_OFF
+        | ClimateEntityFeature.TURN_ON
+    )
+
+    def __init__(
+        self,
+        coordinator: AirPatrolDataUpdateCoordinator,
+        unit: dict[str, Any],
+        unit_id: str,
+    ) -> None:
+        """Initialize the climate entity."""
+        super().__init__(coordinator)
+        self._unit = unit
+        self._unit_id = unit_id
+        unique_id = getattr(coordinator.config_entry, "unique_id", None)
+        self._attr_unique_id = (
+            f"{unique_id}_{unit_id}_climate"
+            if unique_id is not None
+            else f"{unit_id}_climate"
+        )
+        self._attr_translation_key = "climate"
+        self._unavailable_logged = False
+        # Set device info
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unit_id)},
+            name=unit["name"],
+            manufacturer=unit["manufacturer"],
+            model=unit["model"],
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        is_available = (
+            super().available
+            and self.coordinator.data is not None
+            and any(
+                isinstance(u, dict) and u.get("unit_id") == self._unit_id
+                for u in self.coordinator.data
+            )
+            and "climate" in self._unit
+        )
+        if not is_available and not self._unavailable_logged:
+            _LOGGER.info("The climate entity '%s' is unavailable", self._attr_unique_id)
```

Comment:
We should not have this logging as the data update coordinator will do the logging for us if it can't refresh the data.